nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -154,9 +154,20 @@ abstract class NirCodeGen
       else Some(genTypeName(sym.superClass))
 
     def genClassAttrs(sym: Symbol): Attrs = {
-      def pinned =
-        if (!isModule(sym) || isExternModule(sym)) Seq()
-        else Seq(Attr.PinAlways(genMethodName(sym.asClass.primaryConstructor)))
+      def pinned = {
+        val ctor =
+          if (!isModule(sym) || isExternModule(sym)) Seq()
+          else
+            Seq(Attr.PinAlways(genMethodName(sym.asClass.primaryConstructor)))
```

Comment:
Could you tell me why you pin the primary constructor of classes? It might help me understand exactly what pinning does in the context of SN.